src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -1463,6 +1544,10 @@ impl Decimal {
         }
         res
     }
+
+    pub fn is_zero(&self) -> bool {
+        !self.word_buf.iter().any(|&x| x != 0)
```

Comment:
@BusyJay 
If we only set the first byte to 0 in `reset_to_zero`,
we may get false when checking `is_zero(rest_to_zero(123456789987654321.11111))`.

We may set all the bytes to 0 in reset_to_zero (I prefer this.)
or
checking the length of the word_buf array first in is_zero.